src/test/kotlin/dk/cachet/carp/webservices/data/service/CoreDataStreamServiceTest.kt

```diff
@@ -0,0 +1,133 @@
+            every { mutableDataStreamBatch.appendBatch(any()) } answers {
+                println("Append batch called with argument: ${args[0]}")
+                mockk() // Return a mock DataStreamBatch
+            }
+
+            val batch = readBatchFromJson(DATA_STREAM_SERVICE_PATH)
+
+            val data = batch.let { mutableDataStreamBatch.appendBatch(it) }
+
+            verify {
+                mutableDataStreamBatch.appendBatch(batch)
+            }
+            println("Data after appending: $data")
+        }
+
+        @Test
+        fun `should throw if study deployment ID is not matched with dataStreamConfiguration's`() = runTest {
+
+            val studyDeploymentId = UUID.randomUUID()
+            val sut = CoreDataStreamService(
+                mockk(),
+                mockk(),
+                mockk(),
+                objectMapper
+            )
+
+            coEvery { dataStreamConfigurationRepository.findById(studyDeploymentId.stringRepresentation) } returns any()
+
+            val batch = readBatchFromJson(DATA_STREAM_SERVICE_PATH)
+
+            assertFailsWith<IllegalArgumentException> {
+                sut.appendToDataStreams(studyDeploymentId, batch)
+            }
+
+            coVerify(exactly = 0) {
+                dataStreamConfigurationRepository.findById(studyDeploymentId.stringRepresentation)
+            }
+        }
+
+        @Test
+        fun `should throw if study deployment ID does not match with the batch's`() = runTest {
+
+            val studyDeploymentId = UUID.randomUUID()
+
+            val sut = CoreDataStreamService(
+                mockk(),
+                mockk(),
+                mockk(),
+                objectMapper
+            )
+
+            val dataStreamConfigurationRepository = mockk<DataStreamConfigurationRepository>()
+            coEvery { dataStreamConfigurationRepository.findById(studyDeploymentId.stringRepresentation) } returns any()
+
+            val batch = readBatchFromJson(DATA_STREAM_SERVICE_PATH)
+
+            assertThrows<IllegalArgumentException> {
+                sut.appendToDataStreams(studyDeploymentId, batch)
+            }
+
+            coVerify(exactly = 0) {
+                dataStreamConfigurationRepository.findById(any())
+            }
+        }
+
+        private fun readBatchFromJson(jsonFilePath: String): DataStreamBatch {
+            val jsonString = File(javaClass.getResource(jsonFilePath)!!.path).readText()
+            val serialized = json.decodeFromString(DataStreamServiceRequest.Serializer, jsonString)
+
+            val responseSerializer = serialized.getResponseSerializer()
+            println("value of smt: $responseSerializer")
+
+            return when (serialized) {
+                is DataStreamServiceRequest.AppendToDataStreams -> serialized.batch
+                else -> error("Unexpected request type: $serialized")
+            }
```

Comment:
I think you can just pass in the type as a generic type argument for the `decodeFromString` method and you wouldn't need to check the type here.

Reply:
okey